Sources/CMakeLists.txt

```diff
@@ -12,12 +12,29 @@
 ##
 ##===----------------------------------------------------------------------===##
 
-add_subdirectory(_FoundationCShims)
-
-# Disable the macro build on Windows until we can correctly build it for the host architecture
-if(NOT CMAKE_SYSTEM_NAME STREQUAL Windows)
-	add_subdirectory(FoundationMacros)
+include(ExternalProject)
+if(CMAKE_HOST_WIN32)
+  set(_SwiftFlags -DCMAKE_Swift_FLAGS="-use-ld=lld")
+else()
+  set(_SwiftFlags)
+endif()
+ExternalProject_Add(FoundationMacros
+  SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/FoundationMacros"
+  PREFIX "${CMAKE_BINARY_DIR}/_deps"
+  BINARY_DIR "macros"
+  CMAKE_ARGS
+    -DCMAKE_MAKE_PROGRAM=${CMAKE_MAKE_PROGRAM}
+    ${_SwiftFlags}
+  INSTALL_COMMAND "")
+ExternalProject_Get_Property(FoundationMacros BINARY_DIR)
```

Comment:
Will this impact any of the non-macro builds below by changing the value of `BINARY_DIR`?

Reply:
No, `BINARY_DIR` is not the same as `CMAKE_BINARY_DIR` or `CMAKE_CURRENT_BINARY_DIR`.